primitives/api/proc-macro/src/impl_runtime_apis.rs

```diff
@@ -658,20 +699,12 @@ fn generate_runtime_api_versions(impls: &[ItemImpl]) -> Result<TokenStream> {
 		let version: Path = parse_quote!( #path VERSION );
 		let attrs = filter_cfg_attrs(&impl_.attrs);
 
-		result.push(quote!(
-			#( #attrs )*
-			(#id, #version)
-		));
-
-		sections.push(quote!(
-			#( #attrs )*
-			const _: () = {
-				// All sections with the same name are going to be merged by concatenation.
-				#[cfg(not(feature = "std"))]
-				#[link_section = "runtime_apis"]
-				static SECTION_CONTENTS: [u8; 12] = #c::serialize_runtime_api_info(#id, #version);
-			};
-		));
+		if api_ver.is_none() {
+			populate_runtime_api_versions(&mut result, &mut sections, attrs, id, version, &c);
+		} else {
+			let version = api_ver.unwrap();
```

Comment:
This is another thing I plan to rework. Instead of blindly putting the version from the attribute the `decl_runtime_apis!` macro should generate a Vec with all valid versions. One of it needs to be returned here.